optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/loader/InstanceProblemProvider.java

```diff
@@ -57,22 +62,24 @@ public void writeSolution(Solution_ solution, SubSingleBenchmarkResult subSingle
     public boolean equals(Object o) {
         if (this == o) {
             return true;
-        } else if (o instanceof InstanceProblemProvider) {
-            InstanceProblemProvider other = (InstanceProblemProvider) o;
-            return problem.equals(other.problem);
-        } else {
+        }
+        if (!(o instanceof InstanceProblemProvider)) {
             return false;
         }
+        InstanceProblemProvider<?> that = (InstanceProblemProvider<?>) o;
+        return problemName.equals(that.problemName) &&
+                Objects.equals(problem, that.problem) &&
+                Objects.equals(solutionCloner, that.solutionCloner);
     }
 
     @Override
     public int hashCode() {
-        return problem.hashCode();
+        return Objects.hash(problemName, problem, solutionCloner);
     }
 
     @Override
     public String toString() {
-        return problem.toString();
+        return problem != null ? problem.toString() : problemName;
```

Comment:
This should probably just return problemName instead of problem.toString(), already before this change too. wdyt?